ocaml/xapi/repository.ml

```diff
@@ -201,7 +203,12 @@ let sync ~__context ~self ~token ~token_id =
          * I.E. proxy username/password and temporary token file path.
          *)
         write_initial_yum_config ()
-      )
+      ) ;
+    (* The custom yum-utils will fully download repository metadata.*)
+    let repodata_dir =
+      !Xapi_globs.local_pool_repo_dir // repo_name // "repodata" // ".sync_done"
```

Comment:
This looks an implicit expectation on the reposync tool. I understand what we are doing here is to distinguish the two versions of reposync: one can not sync the original metadata, while another can.
I think how about distinguishing by different binaries names?

Reply:
A better option maybe to check the existence of **.asc** file which can distinguish the new version of **reopsync** from the old one easily